go-controller/pkg/ovn/gress_policy.go

```diff
@@ -458,3 +460,71 @@ func (gp *gressPolicy) destroy() error {
 	}
 	return nil
 }
+
+// SVC can be of types 1. clusterIP, 2. NodePort, 3. LoadBalancer,
+// or 4.ExternalIP
+// TODO adjust for upstream patch when it lands:
+// https://bugzilla.redhat.com/show_bug.cgi?id=1908540
+func getSvcVips(service *v1.Service) []net.IP {
+	ips := make([]net.IP, 0)
+
+	if util.ServiceTypeHasNodePort(service) {
+		gatewayRouters, _, err := gateway.GetOvnGateways()
+		if err != nil {
+			klog.Errorf("Cannot get gateways: %s", err)
+		}
+		for _, gatewayRouter := range gatewayRouters {
+			// VIPs would be the physical IPS of the GRs(IPs of the node) in this case
+			physicalIPs, err := gateway.GetGatewayPhysicalIPs(gatewayRouter)
+			if err != nil {
+				klog.Errorf("Unable to get gateway router %s physical ip, error: %v", gatewayRouter, err)
+				continue
+			}
+
+			for _, physicalIP := range physicalIPs {
+				ip := net.ParseIP(physicalIP)
+				if ip == nil {
+					klog.Errorf("Failed to parse pod IP %q", physicalIP)
+					continue
+				}
+				ips = append(ips, ip)
+			}
+		}
+	}
+	if util.ServiceTypeHasClusterIP(service) {
+		if util.IsClusterIPSet(service) {
+			ip := net.ParseIP(service.Spec.ClusterIP)
+			if ip == nil {
+				klog.Errorf("Failed to parse pod IP %q", service.Spec.ClusterIP)
+			}
+			ips = append(ips, ip)
+		}
+
+		for _, ing := range service.Status.LoadBalancer.Ingress {
+			ip := net.ParseIP(ing.IP)
+			if ip == nil {
+				klog.Errorf("Failed to parse pod IP %q", ing)
+				continue
+			}
+			klog.V(5).Infof("Adding ingress IPs from Service: %s to VIP set", service.Name)
+			ips = append(ips, ip)
+		}
+
+		if len(service.Spec.ExternalIPs) > 0 {
+			for _, extIP := range service.Spec.ExternalIPs {
+				ip := net.ParseIP(extIP)
+				if ip == nil {
+					klog.Errorf("Failed to parse pod IP %q", extIP)
+					continue
+				}
+				klog.V(5).Infof("Adding external IPs from Service: %s to VIP set", service.Name)
+				ips = append(ips, ip)
+			}
+		}
+	}
+	if len(ips) == 0 {
+		klog.V(5).Infof("Service has no VIPs")
+		return nil
+	}
+	return ips
```

Comment:
should we check for duplicates?

Reply:
I just moved this function from one of hte files I deleted. @astoycos has plans to remove this entire function so not going to touch it here

Reply:
well since now I am fixing the function I'll fix this too :+1:

Reply:
actually I lied, its not worth it for something we are going to remove soon